### PR TITLE
Backport ignition-diskful and ignition-diskful-subsequent

### PR DESCRIPTION
**Pull request by cgwalters**
Required for https://github.com/openshift/enhancements/pull/210

### CODE REVIEW

File: dracut/30ignition/module-setup.sh [hunk 1]
```diff
@@ -42,18 +51,9 @@ install() {
     inst_simple "$moddir/ignition-generator" \
         "$systemdutildir/system-generators/ignition-generator"
 
-    inst_simple "$moddir/ignition-disks.service" \
-        "$systemdsystemunitdir/ignition-disks.service"
-
     inst_simple "$moddir/ignition-fetch.service" \
```
Comment (bgilbert) [inline dracut/30ignition/module-setup.sh]: Should also be dropped.

File: dracut/30ignition/module-setup.sh [hunk 2]
```diff
@@ -71,11 +71,19 @@ install() {
...
+    install_ignition_unit ignition-mount.service
+    install_ignition_unit ignition-files.service
 
-#   inst_simple "$moddir/coreos-static-network.service" \
-#       "$systemdsystemunitdir/coreos-static-network.service"
+    # units only started when we have a boot disk
+    # path generated by systemd-escape --path /dev/disk/by-label/root
+    install_ignition_unit coreos-gpt-setup@.service        ignition-diskful.target 'coreos-gpt-setup@dev-disk-by\x2dlabel-root.service'
+    install_ignition_unit ignition-setup-user.service      ignition-diskful.target
+    install_ignition_unit ignition-remount-sysroot.service ignition-diskful.target
```
Comment (bgilbert) [inline dracut/30ignition/module-setup.sh]: Still being installed above.  Intentional?

File: dracut/30ignition/module-setup.sh [hunk 3]
```diff
@@ -42,18 +51,9 @@ install() {
...
         "$systemdsystemunitdir/ignition-fetch.service"
 
-    inst_simple "$moddir/ignition-files.service" \
-        "$systemdsystemunitdir/ignition-files.service"
-
-    inst_simple "$moddir/ignition-mount.service" \
-        "$systemdsystemunitdir/ignition-mount.service"
-
     inst_simple "$moddir/ignition-ask-var-mount.service" \
         "$systemdsystemunitdir/ignition-ask-var-mount.service"
```
Comment (bgilbert) [inline dracut/30ignition/module-setup.sh]: Should this be moved into the new target?

Comment (cgwalters) [reply dracut/30ignition/module-setup.sh]: To be clear I only tested this to the point of getting RHCOS rebased on the latest fedora-coreos-config, I didn't yet test it with a live setup. 

Looking at this, probably yes.

**Comment by cgwalters** [comment]
Argh, OK the latest bits get me as far as SELinux denials in the live ISO, but now I broke the mainline case.  Debugging.

**Comment by cgwalters** [comment]
Nevermind, I think that error was something unrelated.  I just tested this again and it works (doesn't break diskful case and gets me a live system except selinux)

File: dracut/30ignition/module-setup.sh [hunk 4]
```diff
@@ -59,6 +59,7 @@ install() {
             "$systemdsystemunitdir/ignition-$x.target"
     done
 
+    install_ignition_unit ignition-setup.service ignition-diskful.target
```
Comment (bgilbert) [inline dracut/30ignition/module-setup.sh]: As currently structured, we're diverging from FCOS by not supporting:

- Per-platform Ignition base configs in live boots (currently unused functionality)
- Supplying an Ignition config via an additional initramfs archive

**Pull request merged**